components/aws/sagemaker/ground_truth/src/sagemaker_ground_truth_component.py

```diff
@@ -0,0 +1,348 @@
+    """SageMaker component for groundtruth."""
+
+    def Do(self, spec: SageMakerGroundTruthSpec):
+        self._labeling_job_name = (
+            spec.inputs.job_name
+            if spec.inputs.job_name
+            else self._generate_unique_timestamped_id(prefix="LabelingJob")
+        )
+        super().Do(spec.inputs, spec.outputs, spec.output_paths)
+
+    def _get_job_status(self) -> SageMakerJobStatus:
+        response = self._sm_client.describe_labeling_job(
+            LabelingJobName=self._labeling_job_name
+        )
+        status = response["LabelingJobStatus"]
+
+        if status == "Completed":
+            return SageMakerJobStatus(
+                is_completed=True, has_error=False, raw_status=status
+            )
+        if status == "Failed":
+            message = response["FailureReason"]
+            return SageMakerJobStatus(
+                is_completed=True,
+                has_error=True,
+                error_message=message,
+                raw_status=status,
+            )
+
+        return SageMakerJobStatus(is_completed=False, raw_status=status)
+
+    def _get_labeling_job_outputs(self, auto_labeling):
+        """Get labeling job output values.
+
+        Args:
+            auto_labeling: Determines whether to return a active learning model ARN.
+
+        Returns:
+            tuple: (Output manifest S3 URI, active learning model ARN)
+        """
+        ### Get and return labeling job outputs
+        info = self._sm_client.describe_labeling_job(
+            LabelingJobName=self._labeling_job_name
+        )
+        output_manifest = info["LabelingJobOutput"]["OutputDatasetS3Uri"]
+        if auto_labeling:
+            active_learning_model_arn = info["LabelingJobOutput"][
+                "FinalActiveLearningModelArn"
+            ]
+        else:
+            active_learning_model_arn = " "
+        return output_manifest, active_learning_model_arn
+
+    def _after_job_complete(
+        self,
+        job: object,
+        request: Dict,
+        inputs: SageMakerGroundTruthInputs,
+        outputs: SageMakerGroundTruthOutputs,
+    ):
+        (
+            outputs.output_manifest_location,
+            outputs.active_learning_model_arn,
+        ) = self._get_labeling_job_outputs(inputs.enable_auto_labeling)
+
+    def _on_job_terminated(self):
+        self._sm_client.stop_labeling_job(LabelingJobName=self._labeling_job_name)
+
+    def _create_job_request(
+        self, inputs: SageMakerGroundTruthInputs, outputs: SageMakerGroundTruthOutputs,
+    ) -> Dict:
+        ### Documentation: https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/sagemaker.html#SageMaker.Client.create_labeling_job
+        request = self._get_request_template("gt")
+
+        # Mapping are extracted from ARNs listed in https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/sagemaker.html#SageMaker.Client.create_labeling_job
+        algorithm_arn_map = {
```

Comment:
Can these come from the `sagemaker` sdk ?

Reply:
I could not find any of these in the `sagemaker` SDK - only found it in the docs.